examples/airflow/dags/hamilton_demo/absenteeism_prediction_dag.py

```diff
@@ -0,0 +1,132 @@
+"""This is an example project using Hamilton with Apache Airflow
+
+For the purpose of this example, we will read and write data from the Airflow
+installation location (${AIRFLOW_HOME}/plugins/data).
+For production environment, use intermediary storage.
+ref: https://docs.astronomer.io/learn/airflow-passing-data-between-tasks
+"""
+
+import os
+from datetime import datetime
+
+from airflow.decorators import dag, task
+from airflow.operators.python import get_current_context
+
+AIRFLOW_HOME = os.getenv("AIRFLOW_HOME")
+
+
+DEFAULT_DAG_PARAMS = dict(
+    raw_data_location=f"{AIRFLOW_HOME}/plugins/data/raw/Absenteeism_at_work.csv",
+    feature_set=[
+        "age_zero_mean_unit_variance",
+        "has_children",
+        "has_pet",
+        "is_summer",
+        "service_time",
+    ],
+    label="absenteeism_time_in_hours",
+    validation_user_ids=[
+        "1",
+        "2",
+        "4",
+        "15",
+        "17",
+        "24",
+        "36",
+    ],
+    h_prepare_data=dict(
+        development_flag=True,
+    ),
+    h_train_and_evaluate_model=dict(
+        development_flag=True,
+        task="binary_classification",
+        pred_path=f"{AIRFLOW_HOME}/plugins/data/results/val_predictions.csv",
+        model_config={},
+        scorer_name="accuracy",
+        bootstrap_iter=1000,
+    ),
+)
+
+
+@dag(
+    dag_id="hamilton-absenteeism-prediction",
+    description="Predict absenteeism using Hamilton and Airflow",
+    start_date=datetime(2023, 6, 18),
+    params=DEFAULT_DAG_PARAMS,
+)
+def absenteeism_prediction_dag():
+    """Predict absenteeism using Hamilton and Airflow"""
+
+    @task
+    def prepare_data():
+        """Load external data, preprocess dataset, and store cleaned data"""
+        # Python imports
+        import pandas as pd
```

Comment:
Airflow does it like that (Metaflow too). Since nodes are ran in isolated environments, it allows to keep node imports to a minimum. Also, a missing packages will only affect the node and won't make the entire Airflow DAG fail / unreadable

https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code